### PR TITLE
Move occupancy sensing related code to dedicated file

### PR DESCRIPTION
**Pull request by SwoopX**
Move any code related to occupancy sensing cluster to a dedicated file. This shall ensure all relevant code is at one place for better overview and to reduce the size of `de_web_plugin.cpp`.

### CODE REVIEW

**Comment by manup** [comment]
Nice one, glad `de_web_plugin.cpp` shrinks a lot lately :+1: 
Since v2.12.1-beta is behind schedule already I'd like to merge this in v2.12.2-beta.

File: occupancy_sensing.cpp [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    {
+        isReadAttr = true;
+    }
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId)
+    {
+        isReporting = true;
+    }
+
+    // Read ZCL reporting and ZCL Read Attributes Response
+    if (isReadAttr || isReporting)
```
Comment (manup) [inline occupancy_sensing.cpp:43]:
```cpp
    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReadAttributesResponseId)
    {
        isReadAttr = true;
    }
    else if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId)
    {
        isReporting = true;
    }
    else
    {
        return; // neigher ZCL Report nor ZCL Read Attributes Response
    }
```

With that we can get rid of the huge *if* scope.

```diff
-    // Read ZCL reporting and ZCL Read Attributes Response
-    if (isReadAttr || isReporting)
-    {
-    }
```

**Review by manup — COMMENTED** [review]
Two small things to consider, while we're at it. Otherwise I'd say go to go

File: occupancy_sensing.cpp [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        }
+    }
+
+    if (!isReporting)
+    {
+        if (!(zclFrame.frameControl() & deCONZ::ZclFCDisableDefaultResponse) && ind.dstAddress().isNwkUnicast())
+        {
+            sendZclDefaultResponse(ind, zclFrame, deCONZ::ZclSuccessStatus);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (SwoopX) [reply occupancy_sensing.cpp]: Indeed not needed anymore

File: occupancy_sensing.cpp (hunk 1 above)
Comment (SwoopX) [reply occupancy_sensing.cpp]: Sure

File: occupancy_sensing.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    QDataStream stream(zclFrame.payload());
+    stream.setByteOrder(QDataStream::LittleEndian);
+
+    bool isReadAttr = false;
+    bool isReporting = false;
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReadAttributesResponseId)
+    {
+        isReadAttr = true;
+    }
+    if (zclFrame.isProfileWideCommand() && zclFrame.commandId() == deCONZ::ZclReportAttributesId)
```
Comment (manup) [inline occupancy_sensing.cpp:37]: Almost there :)

`else if`

Comment (SwoopX) [reply occupancy_sensing.cpp]: Ah, yes. The small little details :)

**Pull request merged**